src/main/java/uk/gov/justice/digital/client/glue/GlueSchemaClient.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.justice.digital.client.glue;
+
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.EntityNotFoundException;
+import com.amazonaws.services.glue.model.GetSchemaVersionRequest;
+import com.amazonaws.services.glue.model.SchemaId;
+import com.amazonaws.services.glue.model.SchemaVersionNumber;
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import lombok.val;
+import uk.gov.justice.digital.config.JobArguments;
+
+import java.util.Optional;
+
+@Singleton
+public class GlueSchemaClient {
+
+    private final AWSGlue glueClient;
+    private final String contractRegistryName;
+
+    @Inject
+    public GlueSchemaClient(GlueClientProvider glueClientProvider,
+                            JobArguments jobArguments) {
+        this.glueClient = glueClientProvider.getClient();
+        this.contractRegistryName = jobArguments.getContractRegistryName();
+    }
+
+    public Optional<String> getSchema(String schemaName) {
```

Comment:
throw the exception to the calling place. Don't catch it in this method

Reply:
This exception is specifically thrown in response to a 404 hence returning an Optional.empty.

Reply:
this is not a correct way of coding. You should call it in the calling place. The method just does what it needs to do, the calling place has to decide the action. We have cutoms exceptions to handle

Reply:
Not in the 404 case. The AWS client throws a specific exception which the caller doesn't have to deal with. This results in cleaner code at the call site.